xds/src/main/java/io/grpc/xds/internal/sds/CommonTlsContextUtil.java

```diff
@@ -27,8 +27,8 @@ private CommonTlsContextUtil() {}
 
   static boolean hasCertProviderInstance(CommonTlsContext commonTlsContext) {
     return commonTlsContext != null
-        && (commonTlsContext.hasTlsCertificateCertificateProviderInstance()
-            || hasCertProviderValidationContext(commonTlsContext));
+        && (hasIdentityCertificateProviderInstance(commonTlsContext)
+        || hasCertProviderValidationContext(commonTlsContext));
```

Comment:
What about
```java
if (commonTlsContext == null) {
  return false;
}
return hasIdentityCertificateProviderInstance(commonTlsContext)
    || hasCertProviderValidationContext(commonTlsContext);
```

to avoid `x && (y || z)`? Ditto to avoid `(x && y) || z`.

Reply:
> ...
> to avoid `x && (y || z)`? Ditto to avoid `(x && y) || z`.

Any particular reason to avoid such expressions? Not aware of any guidelines suggesting that - although the change is easy to make

Reply:
> Any particular reason to avoid such expressions?

I'm feeling the expression is too long when `x,y,z` are long and have parentheses, and therefore hard to parse by human.

Reply:
done